system/HTTP/CURLRequest.php

```diff
@@ -518,18 +518,18 @@ protected function setResponseHeaders(array $headers = [])
     protected function setCURLOptions(array $curlOptions = [], array $config = [])
     {
         // Auth Headers
-        if (! empty($config['auth'])) {
+        if (array_key_exists('auth', $config) && $config['auth']) {
             $curlOptions[CURLOPT_USERPWD] = $config['auth'][0] . ':' . $config['auth'][1];
 
-            if (! empty($config['auth'][2]) && strtolower($config['auth'][2]) === 'digest') {
+            if (array_key_exists(2, $config['auth']) && $config['auth'][2] && strtolower($config['auth'][2]) === 'digest') {
```

Comment:
the second condition should be i think checked against empty string?

Reply:
potensial nullable

Reply:
For this i used checking in function `send()`